specification.md

```diff
@@ -1055,10 +1057,168 @@ For example:
+
+The function can be referenced during workflow execution when the invocation of the REST service is desired. For example:
+
+```json
+{
+  "states":[
+    {
+      "name":"QueryUserInfo",
+      "type":"operation",
+      "actions":[
+        {
+          "functionRef":"queryUserById",
+          "arguments":{
+            "id":"${ .user.id }"
+          }
+        }
+      ],
+      "end":true
+    }
+  ]
+}
+```
+
+Example of the `POST` request sending the state data as part of the body:
+
+```json
+{
+  "functions":[
+    {
+      "name": "createUser",
+      "operation": {
+        "/users": {
+          "post": {
+            "requestBody": {
+              "content": {
+                "application/json": {
+                  "schema": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string"
+                      },
+                      "name": {
+                        "type": "string"                     
+                      },
+                      "email": {
+                        "type": "string"
+                      }
+                    },
+                    "required": ["name", "email"]
+                  }
+                }
+              }
+            }
+          }
+        }
+      },
+      "type": "rest"
+    }
+  ]
+}
+```
+
+Note that the `requestBody` content schema is described inline rather than a reference to an external document.
+
+You can reference the `createUser` function and filter the input data to invoke it, as shown in the example below:
+
+```json
+{
+  "order":{
+    "id":"1234N",
+    "products":[
+      {
+        "name":"Product 1"
+      }
+    ]
+  },
+  "user":{
+    "name":"John Doe",
+    "email":"john@doe.com"
+  }
+}
+```
+
+Function invocation example:
+
+```json
+{
+  "states":[
+    {
+      "name":"CreateNewUser",
+      "type":"operation",
+      "actions":[
+        {
+          "functionRef":"createUser",
+          "actionDataFilter":{
+            "fromStateData":"${ .user }",
+            "toStateData":"${ .user.id }"
+          }
+        }
+      ],
+      "end":true
+    }
+  ]
+}
+```
+
+In this case, only the contents of the `user` attribute will be passed to the function. The user ID returned by the REST request body will then be added to the state data:
+
+```json
+{
+  "order":{
+    "id":"1234N",
+    "products":[
+      {
+        "name":"Product 1"
+      }
+    ]
+  },
+  "user":{
+    "id":"5678U",
+    "name":"John Doe",
+    "email":"john@doe.com"
+  }
+}
+```
+
+The specification does not support the [Security Requirement Object](https://spec.openapis.org/oas/v3.1.0#security-requirement-object). To define authentication for the REST operation, use the [Auth Definition](#Auth-Definition). If provided, this field is ignored.
```

Comment:
Hmmm this is weird. Why dont we use Security Requirement Object. How we answer user question on why?

Reply:
I don't see any info here on validating the inline openapi definitions. Is there anything we can add to tell users that yes we picked to do this cause xyz (like previous comment) and include that its possible to validate (if it really is, idk) something like that would be nice to have

Reply:
> Hmmm this is weird. Why dont we use Security Requirement Object. How we answer user question on why?

Because it's redundant to our `Auth Definition`. I'll add this to the text.

> I don't see any info here on validating the inline openapi definitions. Is there anything we can add to tell users that yes we picked to do this cause xyz (like previous comment) and include that its possible to validate (if it really is, idk) something like that would be nice to have

We can validate based on the import schema we did in the functions schema JSON document (the`$comment` attribute). This way, SDKs and tools can use the OpenAPI spec to validate the operation.